core/common/modules/finder/categories-manager.php

```diff
@@ -34,9 +36,36 @@ class Categories_Manager {
 	 * @access public
 	 * @param string        $category_name
 	 * @param Base_Category $category
+	 *
+	 * @deprecated 3.5.0 - Use `$this->register()`.
 	 */
 	public function add_category( $category_name, Base_Category $category ) {
-		$this->categories[ $category_name ] = $category;
+		Plugin::$instance->modules_manager->get_modules( 'dev-tools' )->deprecation->deprecated_function(
+			__METHOD__,
+			'3.5.0',
+			'register'
+		);
+
+		$this->register( $category, $category_name );
+	}
+
+	/**
+	 * Register a category.
+	 *
+	 * @since 3.5.0
+	 * @access public
+	 * @param Base_Category $instance - Instance of a category.
+	 * @param string        $category_name - Category name - for BC.
+	 *
+	 * @return void
+	 */
+	public function register( Base_Category $instance, $category_name = null ) {
```

Comment:
No `Registrar` class?

Reply:
You mean `Base_Registrar` that every Manager will inherit from? I think that's currently out of the scope of this task.
For now, I just unified the registrations "API" so it'll be easier in the future to create that base class.